numba/tests/test_pycc.py

```diff
@@ -259,10 +259,13 @@ def test_c_extension_usecase(self):
             self.assertPreciseEqual(got, expect)
 
 
-@needs_external_compilers
 class TestDistutilsSupport(TestCase):
 
+
     def setUp(self):
+        super().setUp()
+        self.skip_if_no_external_compiler() # external compiler needed
```

Comment:
I just checked whether it's OK to call `skipTest()` in a `setUp()` method, and it is: https://docs.python.org/3/library/unittest.html#skipping-tests-and-expected-failures